app/models/movie.rb

```diff
@@ -21,40 +21,57 @@ class Movie < ApplicationRecord
   attr_accessor :production_companies
 
   scope :by_title, -> { order(:title) }
-  scope :by_shortest_runtime, -> { order(:runtime) }
-  scope :by_longest_runtime, -> { order(:runtime).reverse_order }
-  scope :by_recent_release_date, -> { order(:release_date).reverse_order }
-  scope :by_highest_vote_average, -> { order(:vote_average).reverse_order }
+  scope :by_shortest_runtime, -> { order(runtime: :asc) }
+  scope :by_longest_runtime, -> { order(runtime: :desc) }
+  scope :by_recent_release_date, -> { order(release_date: :desc) }
+  scope :by_highest_vote_average, -> { order(vote_average: :desc) }
   scope :default_list_order, -> (list) do
-    list.movies.order('CAST(listings.created_at as DATE) desc, listings.priority desc')
+    list.movies.order('listings.created_at DESC, listings.priority DESC')
   end
 
   def self.by_tag_and_user(tag, user)
     joins(:taggings).where(taggings: { user_id: user.id }).where(taggings: { tag_id: tag.id })
   end
 
   def self.by_highest_priority(list)
-    list.movies.sort_by { |movie| movie.priority(list) }.reverse
+    list.movies.order('listings.priority DESC')
   end
 
   def self.by_recently_added(list)
-    list.movies.sort_by { |movie| movie.date_added_to_list(list) }.reverse
+    list.movies.order('listings.created_at DESC')
   end
 
-  def self.by_watched_by_user(list, user)
-    list.movies.sort_by { |movie| movie.viewers.include?(user) ? 0 : 1  }
+  def self.screenings_join_query(user)
+    join_query = <<~SQL.squish
+      LEFT OUTER JOIN screenings
+        ON screenings.movie_id = movies.id
+        AND screenings.user_id = #{user.id}
+    SQL
+    select('movies.*, max(screenings.date_watched) AS max_date_watched')
+      .joins(join_query)
+      .group('movies.id')
   end
```

Comment:
This is the main query used by a few methods so I moved it to its own method.

It allows sorting by unwatched/watched movies, etc. 

It took me a bit to figure it out 😅 .

Basically, left outer joining by movie _and_ user will only populate results if the user has seen the movie.

Grouping by `movies.id` ensures we get distinct results (adding a distinct won't work because of the left outer join).

Reply:
🤓 nice work! This all looks so much better now too! 💪